lib/buy/dfx/dfx_buy_provider.dart

```diff
@@ -152,26 +313,21 @@ class DFXBuyProvider extends BuyProvider {
     }
 
     try {
-      final assetOut = this.assetOut;
-      final blockchain = this.blockchain;
-      final actionType = isBuyAction == true ? '/buy' : '/sell';
+      final actionType = isBuyAction ? '/buy' : '/sell';
 
       final accessToken = await auth();
 
       final uri = Uri.https('services.dfx.swiss', actionType, {
         'session': accessToken,
         'lang': 'en',
-        'asset-out': assetOut,
+        'asset-out': quote.destinationCurrency.toString(),
         'blockchain': blockchain,
-        'asset-in': 'EUR',
+        'asset-in': quote.sourceCurrency.toString(),
+        'amount': amount.toString() //TODO: Amount does not work
```

Comment:
yes, for some reason DFX didn’t see the amount parameter and opened with default ones like 100 EUR or CHF.Fetching quotes and opening a widget are different API services.
Here we are using the wallet address since we need to sign a message, so the user doesn’t need to pass the address. And It’s limiting users a bit because the wallet does not allow fetching DFX quotes if the wallet type does not match the destination cryptocurrency type.